google-cloud-firestore/src/main/java/com/google/cloud/firestore/TransactionOptions.java

```diff
@@ -83,10 +139,169 @@ public static TransactionOptions create(@Nonnull Executor executor) {
    * @param executor The executor to run the user callback code on.
    * @param numberOfAttempts The number of execution attempts.
    * @return The TransactionOptions object.
+   * @deprecated as of 2.0.0, replaced by {@link ReadWriteOptionsBuilder#setExecutor(Executor)} and
+   *     {@link ReadWriteOptionsBuilder#setNumberOfAttempts(int)}
+   * @see #readWriteOptionsBuilder()
    */
   @Nonnull
-  public static TransactionOptions create(@Nonnull Executor executor, int numberOfAttempts) {
-    Preconditions.checkArgument(numberOfAttempts > 0, "You must allow at least one attempt");
-    return new TransactionOptions(numberOfAttempts, executor);
+  @Deprecated
+  public static TransactionOptions create(@Nullable Executor executor, int numberOfAttempts) {
+    return readWriteOptionsBuilder()
+        .setExecutor(executor)
+        .setNumberOfAttempts(numberOfAttempts)
+        .build();
+  }
+
+  @Nonnull
+  public static ReadWriteOptionsBuilder readWriteOptionsBuilder() {
```

Comment:
Just a random thought - feel free to ignore, but could this be `createReadWriteOptionsBuilder()`, which would kind of match the previous API (and break standard naming conventions for builders, which we could fix in the next major version!).

Reply:
Yup, I'll rename the builder factory methods.